Refactor Dokan BOGO integration and fix API endpoint paths

kzamanbd opened the pull request:
Refactor the BOGO integration for improved permissions and correct API endpoint paths in the BogoOffers component to enhance backend interaction.


## Related PR:
* https://github.com/getdokan/dokan/pull/2893

## Closes:
* Closes https://github.com/getdokan/plugin-internal-tasks/issues/901
* Closes https://github.com/getdokan/dokan-pro/issues/4872
* https://github.com/getdokan/storegrowth-sales-booster-pro/issues/169

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a dedicated BOGO area at /bogo with listing and create/edit pages and a "Create New Offer" button.

* **Improvements**
  * Routes moved to store-driven /bogo namespace with pro-gating to hide premium options.
  * Vendor scoping tightened so non-admins see/manage only their own offers.
  * Listing and edit flows show clearer status, product names, pricing, and updated navigation.

* **Bug Fixes**
  * Upgrade overlay now displays only when applicable.

* **Style**
  * Layout and product-name display adjusted for readability.

* **Chores**
  * Settings initialization updated to modern rendering; legacy vendor UI components (product search, older create/edit UIs) removed.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kzamanbd committed:
Refactor Dokan BOGO integration and update permissions

kzamanbd committed:
Fix API endpoint paths in BogoOffers component

Corrected the API endpoint paths for updating offer status and deleting offers in the BogoOffers component to ensure proper interaction with the backend.

kzamanbd committed:
Improve BOGO tab state handling and routing integration

kzamanbd committed:
Update Bogo.php

kzamanbd committed:
Refactor CreateBogo to simplify tab state management

kzamanbd committed:
Merge branch 'develop' into feat/dokan-bogo

kzamanbd committed:
Refactor Bogo integration for pro check and cleanup

kzamanbd committed:
Update BogoOffers.tsx

kzamanbd committed:
Call init_classes directly in Dokan hook registration

kzamanbd committed:
Refactor Dokan dashboard BOGO menu integration

kzamanbd committed:
Add filter for BOGO product query args in Dokan dashboard

kzamanbd committed:
Fix react_route path in Bogo dashboard menu

kzamanbd committed:
Comment out 'Vendors Can Schedule Offers' switcher

kzamanbd committed:
Restrict item queries to creator for non-admin users

mrabbani commented on integrations/assets/src/dokan/dashboard/bogo/index.tsx:
```diff
@@ -1,92 +1,111 @@
-        }
+domReady(() => {
+  // @ts-ignore
+  if (window.spsgAdmin?.isPro) {
```
It may always return `false` since admin scripts are not enqueued.

kzamanbd replied:
localize the `spsgAdmin` on the vendor panel

mrabbani commented on integrations/includes/Dokan/Dashboard/Bogo.php:
```diff
@@ -44,22 +45,32 @@ private function init_hooks() {
             'pos'        => 10,
             'permission' => 'dokandar',
-            'react_route' => 'sales-booster/bogo',
+            'react_route' => 'bogo',
```
Change the route to `stroregrowth/bogo`

kzamanbd replied:
@imtiaz-pranto bhai recommended using only **bogo**. Issue is: https://github.com/getdokan/dokan-pro/issues/4872

mrabbani commented on integrations/includes/Dokan/Dashboard/Bogo.php:
```diff
@@ -79,13 +90,55 @@ public function vendor_dashboard_enqueue_scripts() {
         );
+        wp_localize_script(
+            'spsg-bogo-dokan-vendor-dashboard',
+            'spsgAdmin',
```
Can we change the `spsgAdmin` to `spStoregrowth` since it is not associated with Admin?

kzamanbd replied:
We’re reusing the admin component for **bogo** creation, which is why this prop is required as **spsgAdmin**

mrabbani commented on integrations/includes/Dokan/Dashboard/Bogo.php:
```diff
@@ -79,13 +90,55 @@ public function vendor_dashboard_enqueue_scripts() {
+                'isPro'    => sp_store_growth()->has_pro(),
+            )
+        );
+        $action    = 'ajd_protected';
```
`ajd_` prefix is unknown to me.

kzamanbd replied:
nonce key updated

mrabbani commented on integrations/includes/Dokan/Dashboard/Bogo.php:
```diff
@@ -79,13 +90,55 @@ public function vendor_dashboard_enqueue_scripts() {
+
+        $script = new \StorePulse\StoreGrowth\Modules\BoGo\EnqueueScript();
+
+        wp_localize_script(
```
Pls use [wp_add_inline_script()](https://developer.wordpress.org/reference/functions/wp_add_inline_script/) to localize the script.

mrabbani commented on modules/bogo/includes/REST/BogoController.php:
```diff
@@ -186,6 +186,9 @@ protected function get_total_items_count( $query_filters, $request ) {
     public function get_items( $request ) {
         // Get custom query filters and options from child classes
         $query_filters = $this->get_query_filters( $request );
+		if( ! current_user_can('manage_options') ) {
```
Pls remove this section.

mrabbani replied:
- Use `spsg_bogo_query_args` filter for query
- Use `spsg_bogo_created_by` filter  for vendor despite the staff.

kzamanbd replied:
Added both filters

mrabbani commented on modules/bogo/includes/REST/BogoController.php:
```diff
@@ -615,7 +618,7 @@ public function validate_get_alternate_products( $value, $request, $param ) {
      */
     public function check_permission( $request ) {
-        if ( current_user_can( 'manage_options' ) ) {
+        if ( current_user_can( 'manage_options' ) || current_user_can( 'dokandar' ) ) {
```
Apply filters for permission to enable Dokan logic.

kzamanbd replied:
Filter Added

kzamanbd committed:
Enhance Bogo permissions and query filtering

kzamanbd committed:
Add vendor BOGO enable check for Dokan dashboard

kzamanbd committed:
Update Bogo query filter hooks for REST integration

kzamanbd committed:
Remove vendor enable check in BOGO dashboard

dev-shahed commented:
@kzamanbd  please review the product search box: 
<img width="2346" height="942" alt="image" src="https://github.com/user-attachments/assets/f78577ae-5061-418c-a41d-aa1ca885a802" />

kzamanbd committed:
Add filter for BOGO deal type options and improve compatibility

Introduces a filter ('spsg_bogo_deal_type_options') to allow dynamic modification of BOGO deal type options, with conditional logic based on vendor and pro status. Adds a fallback for useSearchParams to support older Dokan versions, and updates variable naming for clarity in Bogo.php. The BasicInfo component now applies the new filter to its deal type options.
